api/v1beta2/configuration_types.go

```diff
@@ -96,12 +98,378 @@ type Property struct {
 	Value string `json:"value,omitempty"`
 }
 
-// Backend stores the state in a Kubernetes secret with locking done using a Lease resource.
+// Backend describes the Terraform backend configuration
 type Backend struct {
 	// SecretSuffix used when creating secrets. Secrets will be named in the format: tfstate-{workspace}-{secretSuffix}
 	SecretSuffix string `json:"secretSuffix,omitempty"`
 	// InClusterConfig Used to authenticate to the cluster from inside a pod. Only `true` is allowed
 	InClusterConfig bool `json:"inClusterConfig,omitempty"`
+
+	// Inline allows users to use raw hcl code to specify their Terraform backend configuration
+	Inline string `json:"inline,omitempty"`
+
```

Comment:
We need field `BackType` here and its valid values are `remote`, `s3`, `oss`, and so on. it will help to indicate which backend a user chooses.

Reply:
OK

Reply:
Please address this comment.